### PR TITLE
Add Digest extern to PSA/eBPF backend

### PR DESCRIPTION
**Pull request by osinstom**
This PR adds support for Digest extern to PSA/eBPF backend. 

Digests are implemented using `BPF_MAP_TYPE_QUEUE`. An eBPF program generates a Digest by calling `bpf_map_push_elem`. A user space application can get a Digest by calling `psabpf-ctl digest get pipe or `bpf_map_lookup_and_delete_elem` C API. 

This is another PR from series implementing PSA support for eBPF backend.

### CODE REVIEW

File: backends/ebpf/psa/README.md [hunk 1]
```diff
@@ -141,7 +141,17 @@ An inner map is a hash map storing clone session/multicast group members, accord
...
+
+## PSA externs
+
+### Digest
+
+Digests are intended to carry a small piece of user-defined data from the data plane to a control plane.
+The PSA-eBPF compiler translates each Digest instance into `BPF_MAP_TYPE_QUEUE` that implements a FIFO queue.
+If a deparser triggers the `pack()` method, an eBPF program inserts data defined for a Digest into the BPF queue map using `bpf_map_push_elem`.
+A user space application is responsible for performing periodic queries to this map to read a Digest message. It can use either
+`psabpf-ctl digest get pipe` or `bpf_map_lookup_and_delete_elem` C API.
```
Comment (tatry) [inline backends/ebpf/psa/README.md]: `bpf_map_lookup_and_delete_elem` comes from libbpf, might be worth to mention C API from our `psabpf`

File: backends/ebpf/psa/ebpfPsaDeparser.h [hunk 2]
```diff
@@ -30,20 +30,27 @@ class DeparserBodyTranslatorPSA : public DeparserBodyTranslator {
     explicit DeparserBodyTranslatorPSA(const EBPFDeparserPSA* deparser);
 
     void processFunction(const P4::ExternFunction* function) override;
+    void processMethod(const P4::ExternMethod* method) override;
 };
 
 class EBPFDeparserPSA : public EBPFDeparser {
+ private:
+    int maxDigestQueueSize = 100;
```
Comment (tatry) [inline backends/ebpf/psa/ebpfPsaDeparser.h]: Maybe change this to power of 2? (I mean 128) It makes sense?

File: backends/ebpf/tests/ptf/test.py [hunk 3]
```diff
@@ -344,3 +344,21 @@ def runTest(self):
...
+
+    p4_file_path = "p4testdata/digest.p4"
+
+    def runTest(self):
+        pkt = testutils.simple_ip_packet(eth_src="fa:fb:fc:fd:fe:f0")
+        testutils.send_packet(self, PORT0, pkt)
+        testutils.send_packet(self, PORT0, pkt)
+        testutils.send_packet(self, PORT0, pkt)
+
+        digests = self.digest_get("mac_learn_digest_0")
```
Comment (tatry) [inline backends/ebpf/tests/ptf/test.py]: I think that digest instance name should be `mac_learn_digest` or `IngressDeparserImpl_mac_learn_digest` (better - allows to instantiated deparser to use Digest), not a something ended with `_0`

**Comment by kmateuszssak** [comment]
LGTM

**Review by tatry — APPROVED** [review]
LGTM

**Pull request closed without merging**